lib/sisimai/bite/email/apachejames.rb

```diff
@@ -0,0 +1,169 @@
+module Sisimai::Bite::Email
+  module ApacheJames
+    # Sisimai::Bite::::Email::ApacheJames parses a bounce email which created by
+    # ApacheJames. Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/ApacheJames.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :'subject'    => %r/\A\[BOUNCE\]\z/,
+        :'received'   => %r/JAMES SMTP Server/,
+        :'message-id' => %r/\d+[.]JavaMail[.].+[@]/,
+      }
+      Re1 = {
+        # apache-james-2.3.2/src/java/org/apache/james/transport/mailets/
+        #   AbstractNotify.java|124:  out.println("Error message below:");
+        #   AbstractNotify.java|128:  out.println("Message details:");
+        :begin  => %r/\AContent-Disposition:[ ]inline/,
+        :error  => %r/\AError message below:\z/,
+        :rfc822 => %r|\AContent-Type: message/rfc822|,
+        :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Java Apache Mail Enterprise Server'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return []; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Apache James
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        match  = 0
+        match += 1 if mhead['subject'] =~ Re0[:subject]
+        match += 1 if mhead['message-id'] && mhead['message-id'] =~ Re0[:'message-id']
+        match += 1 if mhead['received'].find { |a| a =~ Re0[:received] }
+        return if match.zero?
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [83.06/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->